x-pack/plugins/uptime/public/pages/synthetics/step_detail_page.tsx

```diff
@@ -135,12 +135,18 @@ export const StepDetailPageChildren = () => {
     />
   );
 };
+import { getDynamicSettings } from '../../state/actions/dynamic_settings';
 
 export const StepDetailPage: React.FC = () => {
   useInitApp();
   const { checkGroupId, stepIndex } = useParams<{ checkGroupId: string; stepIndex: string }>();
   useTrackPageview({ app: 'uptime', path: 'stepDetail' });
   useTrackPageview({ app: 'uptime', path: 'stepDetail', delay: 15000 });
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getDynamicSettings());
```

Comment:
Why?

Reply:
we are using data plugin search api, so we need to fetch heartbeat indices settings to construct that query params to pass it to the data.search